06-documenting_R.Rmd

```diff
@@ -0,0 +1,112 @@
+    * Add spaces after commas in the argument lists.
+    * Use `<-` rather than `=` for assignments.
+    * Add spaces around the `<-` operator.
+    * Do not use tabs to indent (as these do not render correctly on all possible pagers).
+    * Use 4 spaces to indent the (example) code.
+    * Make sure the examples are directly executable.
+    * The examples should be system-independent.
+    * The examples should not require special facilities (for instance, Internet access or write permission to specific directories).
+    
+3. `\synopsis` section: The `\usage` section can have the function definition, or the actual definition can be included in the `\synopsis` section.
+
+4. `\source` and `\references` sections:
+    * Author(s) names should be written in the form `Author, A. B.`.
+    * Author(s) names should be separated by a comma or `and` (but not both).
+    * Separate paragraphs (separated by a blank line) should be used for each reference.
+    * Give a date immediatey after the author(s) names.
+    * Do not put a period after the date.
+    * Titles of books and journals (not articles) should be enclosed in `\emph{...}`.
+    * Volume numbers for journals are to be enclosed in `\bold{...}` and followed by a comma.
+    * Use `--` for page ranges.
+    * For given an address for a publisher use the format `New York: Springer-Verlag`.
+
+More guidelines for writing .Rd files can be found [here](https://developer.r-project.org/Rds.html).
+
+The language used in the documentations should follow these basic rules:
+
+1. Affirmative tone should be used to describe what the function does and how to use it effectively. Rather than creating worry in the mind of a reader, it should establish confident knowledge about the effective use of the particular function/feature.
+
+2. More documentation is not necessarily better documentation. Long descriptions full of corner cases and caveats can create the impression that a function is more complex or harder to use than it actually is. Be succinct but exhaustive. 
+
+3. Short code examples can help in understanding better. Readers can often grasp a simple example more quickly than they can digest a formal description. Usually people learn faster with concrete, motivating examples that match the context of a typical use case. 
+
+4. Giving a code equivalent (or approximate equivalent) can be a useful in addition to the description provided. You should carefully weigh whether the code equivalent adds value to the document.
+
+5. The tone of the documentation needs to be respectful of the reader’s background. Lay out the relevant information, show motivating use cases, provide glossary links, and do your best to connect-the-dots. The documentation is meant for newcomers, many of whom will be using it to evaluate the R language as a whole. The experience needs to be positive and not leave the reader with worries that something bad will happen if they make a mistake. 

```

Comment:
I'm not sure about glossary links - what would this look like for an R help file, where would the link go to? Unless we have an example of good practice here, we should probably omit this (not to say that a glossary wouldn't be helpful, but I'm not sure such a thing exists in the core documentation).